Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.h

```diff
@@ -61,6 +62,13 @@ class NumericSettingBase
   // Convert a literal expression e.g. "7.0" to a regular value. (disables expression parsing)
   virtual void SimplifyIfPossible() = 0;
 
+  void RegisterCallback(std::function<void()> function) { m_callback = function; }
```

Comment:
```suggestion
  void RegisterCallback(std::function<void()> function) { m_callback = std::move(function); }
```

Reply:
Ok, I'm new to std::move and really anything with std::function and moving functions around. Can you explain what std::move here does so I can learn? no biggie if you don't want to

Reply:
Yeah, no problem, I'd be glad to:

`std::function` is allowed to allocated memory if it's small internal buffer cannot contain the size of all the captured variables from a lambda exceed the size of it.

Because of that, generally for APIs that take in a `std::function` and assign it unconditionally to an internal member of the same type, you would want to use `std::move` even if that currently weren't the case; since if, through refactoring, that actually happened, you wouldn't need to worry about unnecessary allocations taking place (if that were to actually happen)

Reply:
If anything is unclear, let me know and I can elaborate

Reply:
I think that makes sense, but let me tell you what I understood and you can correct me. `std::function` could potentially allocate heap memory, and using `std::move` transfers the ownership of any heap memory to the member variable without any extra allocations? I assume that helps avoid leaks and improves performance.

Reply:
Yep that's correct!

Reply:
👍